x/leverage/simulation/decoder.go

```diff
@@ -0,0 +1,95 @@
+package simulation
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/kv"
+
+	"github.com/umee-network/umee/x/leverage/types"
+)
+
+// NewDecodeStore returns a decoder function closure that unmarshals the KVPair's
+// Value to the corresponding leverage type.
+func NewDecodeStore(cdc codec.Codec) func(kvA, kvB kv.Pair) string {
+	return func(kvA, kvB kv.Pair) string {
+		prefixA := kvA.Key[:1]
+
+		switch {
+		case bytes.Equal(prefixA, types.KeyPrefixRegisteredToken):
+			var registeredTokenA, registeredTokenB types.Token
+			cdc.MustUnmarshal(kvA.Value, &registeredTokenA)
+			cdc.MustUnmarshal(kvB.Value, &registeredTokenB)
+			return fmt.Sprintf("%v\n%v", registeredTokenA, registeredTokenB)
+
+		case bytes.Equal(prefixA, types.KeyPrefixLoanToken):
+			var loanTokenA, loanTokenB sdk.Coin
+			cdc.MustUnmarshal(kvA.Value, &loanTokenA)
+			cdc.MustUnmarshal(kvB.Value, &loanTokenB)
+			return fmt.Sprintf("%v\n%v", loanTokenA, loanTokenB)
+
+		case bytes.Equal(prefixA, types.KeyPrefixCollateralSetting):
+			return fmt.Sprintf("%v\n%v", kvA, kvB) // it is bytes: []byte{0x01}
+
+		case bytes.Equal(prefixA, types.KeyPrefixCollateralAmount):
+			var collateralAmountA, collateralAmountB sdk.Coin
+			cdc.MustUnmarshal(kvA.Value, &collateralAmountA)
+			cdc.MustUnmarshal(kvB.Value, &collateralAmountB)
+			return fmt.Sprintf("%v\n%v", collateralAmountA, collateralAmountB)
+
+		case bytes.Equal(prefixA, types.KeyPrefixReserveAmount):
+			var collateralAmountA, collateralAmountB sdk.Coin
+			cdc.MustUnmarshal(kvA.Value, &collateralAmountA)
+			cdc.MustUnmarshal(kvB.Value, &collateralAmountB)
+			return fmt.Sprintf("%v\n%v", collateralAmountA, collateralAmountB)
+
+		case bytes.Equal(prefixA, types.KeyPrefixLastInterestTime):
+			lastInterestTimeA, lastInterestTimeB := sdk.ZeroInt(), sdk.ZeroInt()
+			if err := lastInterestTimeA.Unmarshal(kvA.Value); err != nil {
+				panic(fmt.Sprintf("invalid unmarshal value %+v", err))
+			}
+			if err := lastInterestTimeB.Unmarshal(kvB.Value); err != nil {
+				panic(fmt.Sprintf("invalid unmarshal value %+v", err))
+			}
+			return fmt.Sprintf("%v\n%v", lastInterestTimeA, lastInterestTimeB)
```

Comment:
`LastInterestTime` is now being stored as an int64 using `gogotypes.Int64Value` (instead of `sdk.Int`) as seen in `x/leverage/keeper/interest.go` near the bottom of the file. This function should be updated to do the same

Reply:
done at https://github.com/umee-network/umee/pull/401/commits/31a811a288bae32cc0fdf556720a5ea6df9128ad